hypervisor/include/arch/x86/timer.h

```diff
@@ -48,19 +72,57 @@ static inline void initialize_timer(struct hv_timer *timer,
 	}
 }
 
+/**
+ * @brief Check a timer whether expired.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @return true if the timer is expired
+ * @return false if the timer is not expired.
+ */
 static inline bool timer_expired(const struct hv_timer *timer)
 {
 	return ((timer->fire_tsc == 0UL) || (rdtsc() >= timer->fire_tsc));
 }
 
-/*
- * Don't call add_timer/del_timer in the timer callback function.
+/**
+ * @brief Add a timer.
+ *
+ * @param[in] timer Pointer to timer.
+ *
+ * @return 0       - on success.
```

Comment:
I would suggest to use:
```
 * @return
 *       0: on success
 *       \p-EINVAL: \p timer has an invalid value
```

(again, to have a single `return` section in the HTML output)

Reply:
For consistency, this second return value should just be ``\p-EINVAL: timer has an invalid value`` (without the extra ``\p``)